src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

```diff
@@ -0,0 +1,144 @@
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Security;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class TransportWithMessageCredentialNetTCPTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public TransportWithMessageCredentialNetTCPTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact ]
+        public void BasicUserNameAuth()
+        {
+            runTest(false, "testuser@corewcf", _output);
+        }
+
+        [Fact]
+        public void BasicUserNameAuthError()
+        {
+            runTest(true, "randomuser@corewcf", _output);
+        }
+
+        private void runTest(bool isError, String userName, ITestOutputHelper _output)
+        {
+            bool isException = false;
+            try
+            {
+                string testString = new string('a', 3000);
+                IWebHost host = ServiceHelper.CreateWebHostBuilder<StartUpPermissionBaseForTC>(_output).Build();
+                using (host)
+                {
+
+                    host.Start();
+                    System.ServiceModel.NetTcpBinding binding = ClientHelper.GetBufferedModeBinding(System.ServiceModel.SecurityMode.TransportWithMessageCredential);
+                    binding.Security.Message.ClientCredentialType = System.ServiceModel.MessageCredentialType.UserName;
+                    var factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(binding,
+                        new System.ServiceModel.EndpointAddress(host.GetNetTcpAddressInUse() + Startup.WindowsAuthRelativePath));
+                    System.ServiceModel.Description.ClientCredentials clientCredentials = (System.ServiceModel.Description.ClientCredentials)factory.Endpoint.EndpointBehaviors[typeof(System.ServiceModel.Description.ClientCredentials)];
+                    factory.Credentials.ServiceCertificate.SslCertificateAuthentication = new System.ServiceModel.Security.X509ServiceCertificateAuthentication
+                    {
+                        CertificateValidationMode = System.ServiceModel.Security.X509CertificateValidationMode.None
+                    };
+                    clientCredentials.UserName.UserName = userName;
+                    clientCredentials.UserName.Password = RandomString(10);
+                    var channel = factory.CreateChannel();
+                    ((IChannel)channel).Open();
+                    string result = channel.EchoString(testString);
+                    Assert.Equal(testString, result);
+                    ((IChannel)channel).Abort();
+                    factory.Close();
+                }
+            }
+            catch(Exception ex)
+            {
```

Comment:
Assert calls throw an Xunit owned exception. The catch here will catch a throw from Assert.Equal and it will look like the problem was with it being an error. I suggest something more like:
```c#
try
{
    if (isError)
    {
        Assert.Throws<CommunicationException>(() =>
            {
                ((IChannel)channel).Open();
                var result = channel.EchoString(testString);
            });
    }
    else
    {
        ((IChannel)channel).Open();
        var result = channel.EchoString(testString);
        Assert.Equal(testString, result);
    }
}
finally
{
    ((IChannel)channel).Abort();
    factory.Close();
}
```